docs/support-bundle-guide.md

```diff
@@ -0,0 +1,256 @@
+    SupportBundleCollection: true
+```
+
+A single Namespace (e.g., default) is created for saving the Secrets that are
+used to access the support bundle file server, and the permission to read Secrets
+in this Namespace is given to antrea-controller by modifying and applying the
+[RBAC file](../build/yamls/externalnode/support-bundle-collection-rbac.yml).
+
+```yaml
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: antrea-read-secrets
+  namespace: default  # Change the Namespace to where the Secret for file server's authentication credential is created.
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: antrea-secret-reader
+subjects:
+  - kind: ServiceAccount
+    name: antrea-controller
+    namespace: kube-system
+```
+
+## The SupportBundleCollection CRD
+
+SupportBundleCollection CRD is introduced to supplement the `antctl` command
+with three additional features:
+
+1. Allow users to collect support bundle files on external Nodes.
+2. Upload all the support bundle files into a user-provided SFTP Server.
+3. Support tracking status of a SupportBundleCollection CR.
+
+## Usage examples
+
+### Running antctl commands
+
+Please refer to the [antctl user guide section](antctl.md#collecting-support-information).
+Note: `antctl supportbundle` only works on collecting the support bundles from
+Antrea Controller and Antrea Agent that is running on a K8s Node, but it does
+not work for the Agent on an external Node.
+
+### Applying SupportBundleCollection CR
+
+In this section, we will create two SupportBundleCollection CRs for K8s Nodes
+and external Nodes. Note, it is supported to specify Nodes/ExternalNodes by their
+names or by matching their labels in a SupportBundleCollection CR.
+
+Assume we have a cluster with Nodes named "worker1" and "worker2". In addition,
+we have set up two external Nodes named "vm1" and "vm2" in the "vm-ns" Namespace
+by following the instruction of the [VM installation guide](external-node.md#install-antrea-agent-on-vm).
+
+A Secret needs to be created in advance with the username and password of the SFTP
+Server. The Secret will be referred as `authSecret` in the following YAML examples.
+
+```bash
+kubectl create secret generic support-bundle-secret  --from-literal=username='your-sftp-username'  --from-literal=password='your-sftp-password'
+```
+
+Then we can apply the following YAML files. The first one is to collect support
+bundle on K8s Nodes "worker1" and "worker2": "worker1" is specified by the name,
+and "worker2" is specified by matching label "role: workers". The second one is to
+collect support bundle on external Nodes "vm1" and "vm2" in Namespace "vm-ns":
+"vm1" is specified by the name, and "vm2" is specified by matching label "role: vms".
+
+```bash
+cat << EOF | kubectl apply -f -
+kind: SupportBundleCollection
+metadata:
+  name: support-bundle-for-nodes
+spec:
+  nodes:
+    nodeNames: # Empty list means all the Nodes
```

Comment:
When nodeNames is empty but matchLabels is set, does the `nodeName` also means all Nodes? I guess not. Then it's perhaps to move the comment to `nodes:`:
```
nodes:  # All Nodes will be selected if both nodeNames and matchLabels are empty.
```

Reply:
updated.